src/reporting/Nosemgrep.ml

```diff
@@ -128,10 +128,15 @@ let rule_match_nosem ~strict (rule_match : Out.cli_match) :
       let ids = Common.map List.hd (* nosemgrep: list-hd *) ids in
       (* [String.split_on_char] can **not** return an empty list. *)
       (* check if the id specified by the user is the [rule_match]'s [rule_id]. *)
+      let rule_id_last =
```

Comment:
We have a src/core/Rule_ID.mli module we should use for those kinds of things.
There's a function ends_with which you could use I think.

Reply:
Yeah, fair enough. I did this a bit in a rush. Using `ends_with` would accept more things than what "pysemgrep" does, but I'm fine with it.

Reply:
Actually `rule_match.Out.check_id` is just a string, it's not a `Rule_ID.t`.

Reply:
@aryx I can add some `check_nosemgrep : string -> string -> bool` kind of function to `Rule_ID`, but making `Nosemgrep` work with actual `Rule_ID.t` seems out of the scope of this PR. For whatever reason, that module works with `Semgrep_output_v1.cli_match`, and there the `rule_id` is just a plain string.

Reply:
yes I'm working on making the rule_id in cli_match to become a Rule_ID.t (via string wrap;, but it's not yet supported by some atdxxx so I had to pause this refactoring effort).
But I'd rather not inject nosemgrep related stuff in Rule_ID.ml; I would use ends_with(), or maybe make another function like last(), or whatever is needed to make this ruleid: check work.